File: Review/Unit 3/MiniProject/PlaylistObjC/PlaylistObjC/Song.m
//
//  Song.m
//  PlaylistObjC
//
//  Created by Kaleo Kim on 4/21/16.
//  Copyright © 2016 Kaleo Kim. All rights reserved.
//

#import "Song.h"

@implementation Song

-(instancetype)initWithName:(NSString *)name artist:(NSString *)artist {
    
    self = [super init];
    if (self) {
        self.name = name;
        self.artistName = artist;
    }
    return self;
    
}

@end

File: Review/Unit 2/Project/PlaylistMiniProject 2/PlaylistMiniProject/Song.swift
//
//  Song.swift
//  PlaylistMiniProject
//
//  Created by Kaleo Kim on 4/14/16.
//  Copyright © 2016 Brock. All rights reserved.
//

import Foundation
import CoreData


class Song: NSManagedObject {

// Insert code here to add functionality to your managed object subclass
    convenience init(title: String, artist: String, playlist: Playlist, context: NSManagedObjectContext = Stack.sharedStack.managedObjectContext) {
        
        let entity = NSEntityDescription.entity(forEntityName: "Song", in: context)!
        
        self.init(entity: entity, insertInto: context)
        self.title = title
        self.artist = artist
        self.playlist = playlist
    }
}
